test/foxx_test.go

```diff
@@ -0,0 +1,107 @@
+//
+// DISCLAIMER
+//
+// Copyright 2020 ArangoDB GmbH, Cologne, Germany
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Copyright holder is ArangoDB GmbH, Cologne, Germany
+//
+// Author Tomasz Mielech
+//
+package test
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"github.com/arangodb/go-driver"
+	"github.com/stretchr/testify/require"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"testing"
+	"time"
+)
+
+func getZipFile(url, path string) (string, error) {
+	respZip, err := http.Get(url)
+	if err != nil {
+		return "", err
+	}
+	defer respZip.Body.Close()
+
+	zipContent, err := ioutil.ReadAll(respZip.Body)
+	if err != nil {
+		return "", err
+	}
+
+	sha256sum := fmt.Sprintf("%x", sha256.Sum256(zipContent))
+	return sha256sum, ioutil.WriteFile(path, zipContent, 0644)
+}
+
+func TestFoxxItzpapalotlService(t *testing.T) {
+
+	c := createClientFromEnv(t, true)
+	if os.Getenv("TEST_CONNECTION") == "vst" {
+		skipBelowVersion(c, "3.6", t)
+	}
+
+	attempt := 0
+	zipFilePath := "/tmp/itzpapalotl-v1.2.0.zip"
+	for attempt < 3 {
+		sha256sum, err := getZipFile("https://github.com/arangodb-foxx/demo-itzpapalotl/archive/v1.2.0.zip", zipFilePath)
+		require.NoError(t, err)
```

Comment:
Let's wait and see. Sometimes, github is not available. If I understand this correctly, then this will immediately fail. If we observe this too often, we should use the retry loop also for errors and not only for checksum mismatches, and possibly try more times. But we can change this once we see the problem.